### PR TITLE
Add dockerfile to renovate

### PR DESCRIPTION
**Pull request by tangowithfoxtrot**
## Type of change

- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Add Dockerfile to renovate config. Needed to keep the base Rust image up-to-date.

## Code changes

- **.github/renovate.json:** Add rules based on [server repo](https://github.com/bitwarden/server/blob/c0e5d19cb5fc6c44e2110bf4240b4e9e912e6b56/.github/renovate.json#L13-L15
).

## Before you submit

- Please add **unit tests** where it makes sense to do so

### CODE REVIEW

File: .github/renovate.json [hunk 1]
```diff
@@ -2,8 +2,13 @@
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
   "extends": ["github>bitwarden/renovate-config:non-pinned"],
   "separateMajorMinor": true,
-  "enabledManagers": ["cargo", "github-actions", "npm", "nuget"],
+  "enabledManagers": ["cargo", "dockerfile", "github-actions", "npm", "nuget"],
   "packageRules": [
+    {
```
Comment (withinfocus) [inline .github/renovate.json:7]: ⛏️ I prefer these to be alphabetical so it'd go below the Cargo configuration.

**Review by withinfocus — DISMISSED** [review]

File: .github/renovate.json (hunk 1 above)
Comment (tangowithfoxtrot) [reply .github/renovate.json:7]: Me too. Don't know why I missed that.

**Review by withinfocus — APPROVED** [review]

**Review by Hinton — APPROVED** [review]

File: .github/renovate.json [hunk 2]
```diff
@@ -2,14 +2,19 @@
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
   "extends": ["github>bitwarden/renovate-config:non-pinned"],
   "separateMajorMinor": true,
-  "enabledManagers": ["cargo", "github-actions", "npm", "nuget"],
+  "enabledManagers": ["cargo", "dockerfile", "github-actions", "npm", "nuget"],
```
Comment (Hinton) [inline .github/renovate.json:5]:
```suggestion
  "enabledManagers": ["cargo", "docker", "github-actions", "npm", "nuget"],
```

The readme says this should be `docker`? https://docs.renovatebot.com/modules/datasource/docker/

**Review by Hinton — CHANGES_REQUESTED** [review]

**Review by Hinton — APPROVED** [review]
Nvm re-read again and this should be fine.

**Pull request merged**